huaweicloud/services/vpc/resource_huaweicloud_networking_vip.go

```diff
@@ -0,0 +1,275 @@
+	if err != nil {
+		return fmtp.DiagErrorf("Error retrieving HuaweiCloud subnet by network ID (%s): %s", networkId, err)
+	}
+
+	// Check whether the subnet ID entered by the user belongs to the same subnet as the network ID.
+	subnetId := d.Get("subnet_id").(string)
+	if subnetId != "" && subnetId != n.SubnetId && subnetId != n.IPv6SubnetId {
+		return fmtp.DiagErrorf("The subnet ID does not belong to the subnet where the network ID is located.")
+	}
+
+	// Pre-check for subnet network, the virtual IP of IPv6 must be established on the basis that the subnet supports
+	// IPv6.
+	if d.Get("ip_version").(int) == 6 {
+		if n.IPv6SubnetId == "" {
+			return fmtp.DiagErrorf("The subnet does not support IPv6, please enable IPv6 first.")
+		}
+		subnetId = n.IPv6SubnetId
+	} else {
+		subnetId = n.SubnetId
+	}
+
+	opts := ports.CreateOpts{
+		Name:        d.Get("name").(string),
+		DeviceOwner: "neutron:VIP_PORT",
+		NetworkId:   networkId,
+		FixedIps: []ports.FixedIp{
+			{
+				SubnetId:  subnetId,
+				IpAddress: d.Get("ip_address").(string),
+			},
+		},
+	}
+
+	logp.Printf("[DEBUG] Updating network VIP (%s) with options: %#v", d.Id(), opts)
+	vip, err := ports.Create(client, opts)
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud network VIP: %s", err)
+	}
+	logp.Printf("[DEBUG] Waiting for HuaweiCloud network VIP (%s) to become available.", vip.ID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"BUILD"},
+		Target:     []string{"DOWN", "ACTIVE"},
+		Refresh:    waitForNetworkVipStateRefresh(client, vip.ID),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      3 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(vip.ID)
+
+	return resourceNetworkingVipRead(ctx, d, meta)
+}
+
+func setVipNetworkParams(d *schema.ResourceData, port *ports.Port) error {
+	if len(port.FixedIps) > 0 {
+		addr := port.FixedIps[0].IpAddress
+		var ipVersion int
+		if utils.IsIPv4Address(addr) {
+			ipVersion = 4
+		} else {
+			ipVersion = 6
+		}
+		mErr := multierror.Append(nil,
+			d.Set("ip_address", addr),
+			d.Set("subnet_id", port.FixedIps[0].SubnetId),
+			d.Set("ip_version", ipVersion),
+		)
+		return mErr.ErrorOrNil()
+	}
+	return nil
+}
+
+func resourceNetworkingVipRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.NetworkingV1Client(region)
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud VPC network v1 client: %s", err)
+	}
+
+	vip, err := ports.Get(client, d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "VPC network VIP")
+	}
+
+	logp.Printf("[DEBUG] Retrieved VIP %s: %+v", d.Id(), vip)
+
+	d.SetId(vip.ID)
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("status", vip.Status),
```

Comment:
**DOWN** is hard understanding， suggest to convert it to ACTIVE

Reply:
After communicating with relevant personnel, we learned that the status of VIP is always `DOWN`.